src/raftstore/store/peer_storage.rs

```diff
@@ -1090,66 +1165,93 @@ impl PeerStorage {
 
 /// Delete all meta belong to the region. Results are stored in `wb`.
 pub fn clear_meta(
-    engine: &DB,
-    wb: &WriteBatch,
+    kv_wb: &WriteBatch,
+    raft_wb: &WriteBatch,
+    raft_engine: &DB,
     region_id: u64,
     raft_state: &RaftLocalState,
+    is_stale_meta: bool,
 ) -> Result<()> {
     let t = Instant::now();
-    try!(wb.delete(&keys::region_state_key(region_id)));
-
-    let last_index = last_index(raft_state);
-    let mut first_index = last_index + 1;
-    let begin_log_key = keys::raft_log_key(region_id, 0);
-    let end_log_key = keys::raft_log_key(region_id, first_index);
-    try!(engine.scan_cf(
-        CF_RAFT,
-        &begin_log_key,
-        &end_log_key,
-        false,
-        &mut |key, _| {
-            first_index = keys::raft_log_index(key).unwrap();
-            Ok(false)
+    try!(kv_wb.delete(&keys::region_state_key(region_id)));
+    try!(kv_wb.delete(&keys::apply_state_key(region_id)));
+
+    let raft_count = if is_stale_meta {
+        try!(clear_raft_data(raft_wb, raft_engine, region_id))
+    } else {
+        let last_index = last_index(raft_state);
```

Comment:
What if a pending log gc task isn't executed in time when it's about to be destroyed?